sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -737,6 +738,8 @@ private String parentResourceLinkToQueryLink(String parentResourceLink, Resource
             IDocumentQueryClient queryClient,
             UUID activityId) {
 
+        SqlQuerySpecLogger.getInstance().logQuery(sqlQuery);
```

Comment:
@mbhaskar is this the right place?

Reply:
`queryDocuments` would be a better place IMHO if all we want is to log the document/item queries. 
https://github.com/Azure/azure-sdk-for-java/blob/e01f66b2c9b4637e98cdbf1a247745ac439b388e/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java#L2204

Placing logging in `createQueryInternal` would log various other queries including databases/collections etc, which may not be the intention here.

Reply:
makes sense, moved